test-basic/documents-data-movement-readAll.js

```diff
@@ -0,0 +1,236 @@
+const streamToArray = require('stream-to-array');
+let uriStream = new Stream.Readable();
+let urisList = [];
+let result = new Set();
+let summaryValue = null;
+
+describe('data movement readAll', function() {
+    before(function (done) {
+        let readable = new Stream.Readable({objectMode: true});
+        for(let i=0; i<10000; i++) {
+            const temp = {
+                uri: '/test/dataMovement/requests/readAll/'+i+'.json',
+                contentType: 'application/json',
+                content: {['key '+i]:'value '+i}
+            };
+            readable.push(temp);
+            urisList.push(temp.uri);
+        }
+        readable.push(null);
+        readable.pipe(dbWriter.documents.writeAll());
+        setTimeout(()=>{done();}, 3000);
+    });
+
+    beforeEach(function(done){
+        uriStream = new Stream.PassThrough({objectMode: true});
+        urisList.forEach(uri => uriStream.push(uri));
+        setTimeout(()=>{uriStream.push(null);done();}, 3000);
+    });
+
+    after((function(done){
+        dbWriter.documents.remove(urisList)
+            .result(function(){
+                done();
+            })
+            .catch(err=> done(err))
+            .catch(done);
+    }));
+
+    it('should readAll documents with empty options', function(done){
+
+        streamToArray(uriStream.pipe(dbWriter.documents.readAll()),
+            function(err, arr ) {
+                if(err){
+                    done(err);
+                }
+                arr.forEach(item=> result.add(item.uri)
+                );
+                checkResult(done);
+            });
+    });
+
+    it('should readAll documents with onCompletion option', function(done){
+
+        streamToArray(uriStream.pipe(dbWriter.documents.readAll({
+                onCompletion: ((summary) => summaryValue = summary)
+        })),
+            function(err, arr ) {
+                if(err){
+                    done(err);
+                }
+                arr.forEach(item=> result.add(item.uri));
+                checkSummary(summaryValue, done);
+            });
+    });
+
+    it('should throw error with invalid inputKind option', function(done){
+        try{
+            dbWriter.documents.readAll({inputKind:10});
+        } catch(err){
+            err.toString().should.equal('Error: Invalid value for inputKind. Value must be array or string.');
+            done();
+        }
+    });
+
+    it('should readAll documents with inputKind option as string', function(done){
+        streamToArray(uriStream.pipe(dbWriter.documents.readAll({
+                inputKind: 'String'
+            })),
+            function(err, arr ) {
+                if(err){
+                    done(err);
+                }
+                arr.forEach(item=> result.add(item.uri)
+                );
+                checkResult(done);
+            });
+    });
+
+    it('should readAll documents with inputKind option as array', function(done){
```

Comment:
A good test might be to 

- construct a readable stream in object mode
- iterate over urislist in increments of 10
- on each iteration, call slice(i, i + 10) to construct an array of 10
- add each array of 10 as a single item to the readable stream
- pipe the readable stream to readAll()